### PR TITLE
Sync Package: Moves the Protect sync module

### PR DESCRIPTION
**Pull request by roccotripaldi**
We are trying to move away from class mapped packages, and unspool all of the dependencies within a package.

So let's do this!

Part of #12712

#### Changes proposed in this Pull Request:

This PR should have no functional changes. The Protect sync module should function exactly the same as before.

Note, the Protect module will need its own package, but we'll handle that in an other PR.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

This modifies an existing part of Jetpack.

#### Testing instructions:

* Try this branch out on a Jurassic.ninja site or on your docker testing site.
* Connect the site and activate the recommended features.
* Point your testing site to your .com sandbox to observe sync actions in real time.
* Log out of your testing site, and login with a incorrect username / password
* Do you see the sync action for `jetpack_valid_failed_login_attempt`

#### Proposed changelog entry for your changes:

*

### CODE REVIEW

File: packages/sync/legacy/class.jetpack-sync-modules.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ class Jetpack_Sync_Modules {
 		'Jetpack_Sync_Module_Users',
 		'Jetpack_Sync_Module_Posts',
 		'Jetpack_Sync_Module_Import',
-		'Jetpack_Sync_Module_Protect',
+		'Automattic\\Jetpack\\Sync\\Modules\\Protect',
```
Comment (gravityrail) [inline packages/sync/legacy/class.jetpack-sync-modules.php]: Probably ought to be namespaced inside a package for the Protect module itself, like

`\Automattic\Jetpack\Protect\Sync_Module`

Then the protect Sync module should be registered when the Protect module is active - right?

File: composer.lock [hunk 2]
```diff
@@ -8,12 +8,11 @@
     "packages": [
         {
             "name": "automattic/jetpack-asset-tools",
-            "version": "dev-master",
+            "version": "dev-add/protect-module-sync-package",
```
Comment (lezama) [inline composer.lock]: we shouldn't merge these changes to the version

File: packages/sync/legacy/class.jetpack-sync-modules.php (hunk 1 above)
Comment (roccotripaldi) [reply packages/sync/legacy/class.jetpack-sync-modules.php]: I'm going to handle that in an other PR

File: composer.lock (hunk 2 above)
Comment (roccotripaldi) [reply composer.lock]: @lezama - how do i fix this?

Comment (lezama) [reply composer.lock]: manually is how I have seen @tyxla  doing it, replacing `dev-add/protect-module-sync-package` with `dev-master`

File: composer.lock [hunk 3]
```diff
@@ -12,8 +12,7 @@
             "dist": {
                 "type": "path",
                 "url": "./packages/asset-tools",
-                "reference": "5733de68bc0616f37ed3813b092908767df2cf68",
```
Comment (lezama) [inline composer.lock]: let's not push these

**Review by gititon — DISMISSED** [review]

File: packages/sync/legacy/class.jetpack-sync-module-protect.php [hunk 4]
```diff
@@ -1,25 +0,0 @@
-<?php
-
-use Automattic\Jetpack\Constants;
```
Comment (lezama) [inline packages/sync/legacy/class.jetpack-sync-module-protect.php]: let's make sure we `git mv` instead!

**Pull request closed without merging**